examples/mixed-spmv/mixed-spmv.cpp

```diff
@@ -0,0 +1,271 @@
+        std::cerr << "Usage: " << argv[0] << " [executor] " << std::endl;
+        std::exit(-1);
+    }
+
+    // @sect3{Where do you want to run your operation?}
+    // The gko::Executor class is one of the cornerstones of Ginkgo. Currently,
+    // we have support for
+    // an gko::OmpExecutor, which uses OpenMP multi-threading in most of its
+    // kernels, a gko::ReferenceExecutor, a single threaded specialization of
+    // the OpenMP executor and a gko::CudaExecutor which runs the code on a
+    // NVIDIA GPU if available.
+    // @note With the help of C++, you see that you only ever need to change the
+    // executor and all the other functions/ routines within Ginkgo should
+    // automatically work and run on the executor with any other changes.
+    const auto executor_string = argc >= 2 ? argv[1] : "reference";
+    std::map<std::string, std::function<std::shared_ptr<gko::Executor>()>>
+        exec_map{
+            {"omp", [] { return gko::OmpExecutor::create(); }},
+            {"cuda",
+             [] {
+                 return gko::CudaExecutor::create(0, gko::OmpExecutor::create(),
+                                                  true);
+             }},
+            {"hip",
+             [] {
+                 return gko::HipExecutor::create(0, gko::OmpExecutor::create(),
+                                                 true);
+             }},
+            {"dpcpp",
+             [] {
+                 return gko::DpcppExecutor::create(0,
+                                                   gko::OmpExecutor::create());
+             }},
+            {"reference", [] { return gko::ReferenceExecutor::create(); }}};
+
+    // executor where Ginkgo will perform the computation
+    const auto exec = exec_map.at(executor_string)();  // throws if not valid
+
+    // @sect3{Preparing your data and transfer to the proper device.}
+    // Read the matrix using the @ref read function and set the right hand side
+    // randomly.
+    // @note Ginkgo uses C++ smart pointers to automatically manage memory. To
+    // this end, we use our own object ownership transfer functions that under
+    // the hood call the required smart pointer functions to manage object
+    // ownership. The gko::share , gko::give and gko::lend are the functions
+    // that you would need to use.
+
+    // read the matrix into HighPrecision and LowPrecision.
+    auto hp_A = share(gko::read<hp_mtx>(std::ifstream("data/A.mtx"), exec));
+    auto lp_A = share(gko::read<lp_mtx>(std::ifstream("data/A.mtx"), exec));
+    auto host_b =
+        hp_vec::create(exec->get_master(), gko::dim<2>{hp_A->get_size()[1], 1});
+    std::ranlux48 rand_engine(30);
+    auto dist = std::uniform_real_distribution<RealValueType>(0.0, 1.0);
+    for (int i = 0; i < host_b->get_size()[0]; i++) {
+        host_b->get_values()[i] =
+            get_rand_value<HighPrecision>(dist, rand_engine);
+    }
+    auto hp_b = share(hp_vec::create(exec));
```

Comment:
I think here we could also add a few comments explaining what you are doing? Creating different vectors on the device